登录不上，滑动验证码后，提示网络环境异常，

登录不上，滑动验证码后，提示网络环境异常，

我来追加我这遇到的情况：↓
**mirai-console**: mirai-console-2.0-M1-1-all.jar
**mirai-console-terminal**: mirai-console-terminal-2.0-M1-1-all.jar
**mirai-core**: mirai-core-all-2.0-M1-1-all.jar
**设备所状态**: 开启设备锁与关闭设备锁均提示禁止登录
**尝试登录次数**: 具体次数记不清了，大概尝试了 12 次，均无法登录
**第一次尝试登录与最后一次尝试登录的间隔时间**: 大于 5 小时

```
2020-12-21 14:04:51 I/main: Starting mirai-console...
2020-12-21 14:04:51 I/main: Backend: version 2.0-M1-1, built on 2020-12-19 23:26:13.
2020-12-21 14:04:51 I/main: Frontend Terminal: version 2.0-M1-1, provided by Mamoe Technologies
2020-12-21 14:04:51 I/plugin: Successfully loaded plugin net.mamoe.mirai-api-http
2020-12-21 14:04:51 I/main: Prepared built-in commands: autoLogin, help, login, permission, status, stop
2020-12-21 14:04:51 W/net.mamoe.mirai-api-http: USING INITIAL KEY, please edit the key
2020-12-21 14:04:51 I/net.mamoe.mirai-api-http: Starting Mirai HTTP Server in 0.0.0.0:8080
2020-12-21 14:04:51 I/Mirai HTTP API: Http api server is running with authKey: INITKEYOeMLf1Av
2020-12-21 14:04:51 I/net.mamoe.mirai-api-http: 心跳模块启用状态: false
2020-12-21 14:04:52 I/net.mamoe.mirai-api-http: 上报模块启用状态: false
2020-12-21 14:04:52 I/main: 1 plugin(s) enabled.
2020-12-21 14:04:52 I/main: mirai-console started successfully.
> login xxxxx xxxxxxx
2020-12-21 14:05:10 I/Mirai: Mirai 正在使用桌面环境. 如遇到验证码将会弹出对话框. 可添加 JVM 属性 `mirai.no-desktop` 以关闭.
2020-12-21 14:05:10 I/Mirai: Mirai is using desktop. Captcha will be thrown by window popup. You can add `mirai.no-desktop` to JVM properties (-Dmirai.no-desktop) to disable it.
2020-12-21 14:05:12 W/stderr: java.security.NoSuchProviderException: JCE cannot authenticate the provider BC
2020-12-21 14:05:12 W/stderr:   at java.base/javax.crypto.JceSecurity.getInstance(JceSecurity.java:132)
2020-12-21 14:05:12 W/stderr:   at java.base/javax.crypto.KeyAgreement.getInstance(KeyAgreement.java:238)
2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.internal.utils.crypto.ECDH$Companion.calculateShareKey(ECDHJvmDesktop.kt:81)
2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.internal.utils.crypto.ECDH$Companion$1$1.invoke(ECDHJvmDesktop.kt:49)
2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.internal.utils.crypto.ECDH.<clinit>(ECDHJvmDesktop.kt:61)
2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.internal.utils.crypto.ECDHJvmDesktopKt.ECDH(ECDHJvmDesktop.kt:34)
2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.internal.network.QQAndroidClient.<init>(QQAndroidClient.kt:75)
2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.internal.QQAndroidBot.<init>(QQAndroidBot.kt:48)
2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.internal.BotFactoryImpl.newBot(BotFactory.kt:30)
2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.BotFactory$INSTANCE.newBot(BotFactory.kt:74)
2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.console.MiraiConsole$INSTANCE.addBotImpl(MiraiConsole.kt:161)
2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.console.MiraiConsole$INSTANCE.addBot(MiraiConsole.kt:125)
2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.console.MiraiConsole$INSTANCE.addBot$default(MiraiConsole.kt:124)
2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.console.command.BuiltInCommands$LoginCommand.handle(BuiltInCommands.kt:158)
2020-12-21 14:05:12 W/stderr:   at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2020-12-21 14:05:12 W/stderr:   at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2020-12-21 14:05:12 W/stderr:   at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2020-12-21 14:05:12 W/stderr:   at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2020-12-21 14:05:12 W/stderr:   at kotlin.reflect.jvm.internal.calls.CallerImpl$Method.callMethod(CallerImpl.kt:97)
2020-12-21 14:05:12 W/stderr:   at kotlin.reflect.jvm.internal.calls.CallerImpl$Method$Instance.call(CallerImpl.kt:113)
2020-12-21 14:05:12 W/stderr:   at kotlin.reflect.jvm.internal.KCallableImpl.call(KCallableImpl.kt:108)
2020-12-21 14:05:12 W/stderr:   at kotlin.reflect.jvm.internal.KCallableImpl.callDefaultMethod$kotlin_reflection(KCallableImpl.kt:159)
2020-12-21 14:05:12 W/stderr:   at kotlin.reflect.full.KCallables.callSuspendBy(KCallables.kt:73)
2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.console.internal.command.CommandReflector$findSubCommands$6$1.invokeSuspend(CommandReflector.kt:283)
2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.console.internal.command.CommandReflector$findSubCommands$6$1.invoke(CommandReflector.kt)
2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.console.command.descriptor.CommandSignatureFromKFunctionImpl.call$suspendImpl(CommandSignature.kt:88)
2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.console.command.descriptor.CommandSignatureFromKFunctionImpl.call(CommandSignature.kt)
2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.console.internal.command.CommandManagerImplKt.executeCommandImpl(CommandManagerImpl.kt:161)
2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.console.command.CommandManager.executeCommand(CommandManager.kt:129)
2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.console.command.CommandManager$INSTANCE.executeCommand(CommandManager.kt)
2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.console.terminal.ConsoleThreadKt$startupConsoleThread$3.invokeSuspend(ConsoleThread.kt:183)
2020-12-21 14:05:12 W/stderr:   at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
2020-12-21 14:05:12 W/stderr:   at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
2020-12-21 14:05:12 W/stderr:   at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
2020-12-21 14:05:12 W/stderr:   at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:738)
2020-12-21 14:05:12 W/stderr:   at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
2020-12-21 14:05:12 W/stderr:   at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
2020-12-21 14:05:12 W/stderr: Caused by: java.util.jar.JarException: file:/C:/Users/Administrator/Desktop/mirai/libs/mirai-core-all-2.0-M1-1-all.jar has unsigned entries - net/mamoe/mirai/internal/AbstractBot$2.class
2020-12-21 14:05:12 W/stderr:   at java.base/javax.crypto.JarVerifier.verifySingleJar(JarVerifier.java:459)
2020-12-21 14:05:12 W/stderr:   at java.base/javax.crypto.JarVerifier.verifyJars(JarVerifier.java:314)
2020-12-21 14:05:12 W/stderr:   at java.base/javax.crypto.JarVerifier.verify(JarVerifier.java:257)
2020-12-21 14:05:12 W/stderr:   at java.base/javax.crypto.ProviderVerifier.verify(ProviderVerifier.java:129)
2020-12-21 14:05:12 W/stderr:   at java.base/javax.crypto.JceSecurity.verifyProvider(JceSecurity.java:191)
2020-12-21 14:05:12 W/stderr:   at java.base/javax.crypto.JceSecurity.getVerificationResult(JceSecurity.java:217)
2020-12-21 14:05:12 W/stderr:   at java.base/javax.crypto.JceSecurity.getInstance(JceSecurity.java:129)
2020-12-21 14:05:12 W/stderr:   ... 36 more
2020-12-21 14:05:12 I/Bot.xxxxxxxxxxxx: Logging in...
2020-12-21 14:05:12 I/Bot.xxxxxxxxxxxx: 需要滑动验证码
2020-12-21 14:05:12 I/Bot.xxxxxxxxxxxx: 请在任意浏览器中打开以下链接并完成验证码.
2020-12-21 14:05:12 I/Bot.xxxxxxxxxxxx: 完成后请输入任意字符
2020-12-21 14:05:12 I/Bot.xxxxxxxxxxxx: https://xxxxxxxxxxxxxxxxxxxxxxx.............................
14:05:12 LOGIN>
LOGIN>  > 1
2020-12-21 14:05:24 I/Bot.xxxxxxxxxxxx: 正在提交中...
Login failed: Error(title=禁止登录, message=当前上网环境异常，请更换网络环境或在常用设备上登录或稍后再试。, errorInfo=)
2020-12-21 14:05:24 E/console: net.mamoe.mirai.network.WrongPasswordException: Error(title=禁止登录, message=当前上网环 境异常，请更换网络环境或在常用设备上登录或稍后再试。, errorInfo=)
net.mamoe.mirai.network.WrongPasswordException: Error(title=禁止登录, message=当前上网环境异常，请更换网络环境或在常用设备上登录或稍后再试。, errorInfo=)
        at net.mamoe.mirai.internal.network.QQAndroidBotNetworkHandler.closeEverythingAndRelogin(QQAndroidBotNetworkHandler.kt:185)
        at net.mamoe.mirai.internal.network.QQAndroidBotNetworkHandler$closeEverythingAndRelogin$1.invokeSuspend(QQAndroidBotNetworkHandler.kt)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.internal.ScopeCoroutine.afterResume(Scopes.kt:32)
        at kotlinx.coroutines.AbstractCoroutine.resumeWith(AbstractCoroutine.kt:113)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:738)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)

> login xxxxxxxxxxxx xxxxxxx
2020-12-21 14:11:42 I/Bot.xxxxxxxxxxxx: Logging in...
2020-12-21 14:11:42 I/Bot.xxxxxxxxxxxx: 需要滑动验证码
2020-12-21 14:11:42 I/Bot.xxxxxxxxxxxx: 请在任意浏览器中打开以下链接并完成验证码.
2020-12-21 14:11:42 I/Bot.xxxxxxxxxxxx: 完成后请输入任意字符
2020-12-21 14:11:42 I/Bot.xxxxxxxxxxxx: https://xxxxxxxxxxxxxxxxxxxxxxx.............................
14:11:42 LOGIN>
LOGIN>  >
2020-12-21 14:12:28 I/Bot.xxxxxxxxxxxx: 正在提交中...
Login failed: Error(title=禁止登录, message=当前上网环境异常，请更换网络环境或在常用设备上登录或稍后再试。, errorInfo=)
2020-12-21 14:12:28 E/console: net.mamoe.mirai.network.WrongPasswordException: Error(title=禁止登录, message=当前上网环 境异常，请更换网络环境或在常用设备上登录或稍后再试。, errorInfo=)
net.mamoe.mirai.network.WrongPasswordException: Error(title=禁止登录, message=当前上网环境异常，请更换网络环境或在常用设备上登录或稍后再试。, errorInfo=)
        at net.mamoe.mirai.internal.network.QQAndroidBotNetworkHandler.closeEverythingAndRelogin(QQAndroidBotNetworkHandler.kt:185)
        at net.mamoe.mirai.internal.network.QQAndroidBotNetworkHandler$closeEverythingAndRelogin$1.invokeSuspend(QQAndroidBotNetworkHandler.kt)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.internal.ScopeCoroutine.afterResume(Scopes.kt:32)
        at kotlinx.coroutines.AbstractCoroutine.resumeWith(AbstractCoroutine.kt:113)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:738)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
```

先换上openjdk再说（


 
---原始邮件---
发件人: "ieew"<notifications@github.com&gt;
发送时间: 2020年12月21日 14:37:51
收件人: "mamoe/mirai"<mirai@noreply.github.com&gt;;
抄送: "Subscribed"<subscribed@noreply.github.com&gt;;
主题: Re: [mamoe/mirai] 登录不上，滑动验证码后，提示网络环境异常， (#745)




 
我来追加我这遇到的情况：↓
 mirai-console: mirai-console-2.0-M1-1-all.jar
 mirai-console-terminal: mirai-console-terminal-2.0-M1-1-all.jar
 mirai-core: mirai-core-all-2.0-M1-1-all.jar
 设备所状态: 开启设备锁与关闭设备锁均提示禁止登录
 尝试登录次数: 具体次数记不清了，大概尝试了 12 次，均无法登录
 第一次尝试登录与最后一次尝试登录的间隔时间: 大于 5 小时
 2020-12-21 14:04:51 I/main: Starting mirai-console... 2020-12-21 14:04:51 I/main: Backend: version 2.0-M1-1, built on 2020-12-19 23:26:13. 2020-12-21 14:04:51 I/main: Frontend Terminal: version 2.0-M1-1, provided by Mamoe Technologies 2020-12-21 14:04:51 I/plugin: Successfully loaded plugin net.mamoe.mirai-api-http 2020-12-21 14:04:51 I/main: Prepared built-in commands: autoLogin, help, login, permission, status, stop 2020-12-21 14:04:51 W/net.mamoe.mirai-api-http: USING INITIAL KEY, please edit the key 2020-12-21 14:04:51 I/net.mamoe.mirai-api-http: Starting Mirai HTTP Server in 0.0.0.0:8080 2020-12-21 14:04:51 I/Mirai HTTP API: Http api server is running with authKey: INITKEYOeMLf1Av 2020-12-21 14:04:51 I/net.mamoe.mirai-api-http: 心跳模块启用状态: false 2020-12-21 14:04:52 I/net.mamoe.mirai-api-http: 上报模块启用状态: false 2020-12-21 14:04:52 I/main: 1 plugin(s) enabled. 2020-12-21 14:04:52 I/main: mirai-console started successfully. &gt; login xxxxx xxxxxxx 2020-12-21 14:05:10 I/Mirai: Mirai 正在使用桌面环境. 如遇到验证码将会弹出对话框. 可添加 JVM 属性 `mirai.no-desktop` 以关闭. 2020-12-21 14:05:10 I/Mirai: Mirai is using desktop. Captcha will be thrown by window popup. You can add `mirai.no-desktop` to JVM properties (-Dmirai.no-desktop) to disable it. 2020-12-21 14:05:12 W/stderr: java.security.NoSuchProviderException: JCE cannot authenticate the provider BC 2020-12-21 14:05:12 W/stderr:   at java.base/javax.crypto.JceSecurity.getInstance(JceSecurity.java:132) 2020-12-21 14:05:12 W/stderr:   at java.base/javax.crypto.KeyAgreement.getInstance(KeyAgreement.java:238) 2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.internal.utils.crypto.ECDH$Companion.calculateShareKey(ECDHJvmDesktop.kt:81) 2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.internal.utils.crypto.ECDH$Companion$1$1.invoke(ECDHJvmDesktop.kt:49) 2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.internal.utils.crypto.ECDH.<clinit&gt;(ECDHJvmDesktop.kt:61) 2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.internal.utils.crypto.ECDHJvmDesktopKt.ECDH(ECDHJvmDesktop.kt:34) 2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.internal.network.QQAndroidClient.<init&gt;(QQAndroidClient.kt:75) 2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.internal.QQAndroidBot.<init&gt;(QQAndroidBot.kt:48) 2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.internal.BotFactoryImpl.newBot(BotFactory.kt:30) 2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.BotFactory$INSTANCE.newBot(BotFactory.kt:74) 2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.console.MiraiConsole$INSTANCE.addBotImpl(MiraiConsole.kt:161) 2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.console.MiraiConsole$INSTANCE.addBot(MiraiConsole.kt:125) 2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.console.MiraiConsole$INSTANCE.addBot$default(MiraiConsole.kt:124) 2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.console.command.BuiltInCommands$LoginCommand.handle(BuiltInCommands.kt:158) 2020-12-21 14:05:12 W/stderr:   at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) 2020-12-21 14:05:12 W/stderr:   at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) 2020-12-21 14:05:12 W/stderr:   at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) 2020-12-21 14:05:12 W/stderr:   at java.base/java.lang.reflect.Method.invoke(Method.java:566) 2020-12-21 14:05:12 W/stderr:   at kotlin.reflect.jvm.internal.calls.CallerImpl$Method.callMethod(CallerImpl.kt:97) 2020-12-21 14:05:12 W/stderr:   at kotlin.reflect.jvm.internal.calls.CallerImpl$Method$Instance.call(CallerImpl.kt:113) 2020-12-21 14:05:12 W/stderr:   at kotlin.reflect.jvm.internal.KCallableImpl.call(KCallableImpl.kt:108) 2020-12-21 14:05:12 W/stderr:   at kotlin.reflect.jvm.internal.KCallableImpl.callDefaultMethod$kotlin_reflection(KCallableImpl.kt:159) 2020-12-21 14:05:12 W/stderr:   at kotlin.reflect.full.KCallables.callSuspendBy(KCallables.kt:73) 2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.console.internal.command.CommandReflector$findSubCommands$6$1.invokeSuspend(CommandReflector.kt:283) 2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.console.internal.command.CommandReflector$findSubCommands$6$1.invoke(CommandReflector.kt) 2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.console.command.descriptor.CommandSignatureFromKFunctionImpl.call$suspendImpl(CommandSignature.kt:88) 2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.console.command.descriptor.CommandSignatureFromKFunctionImpl.call(CommandSignature.kt) 2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.console.internal.command.CommandManagerImplKt.executeCommandImpl(CommandManagerImpl.kt:161) 2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.console.command.CommandManager.executeCommand(CommandManager.kt:129) 2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.console.command.CommandManager$INSTANCE.executeCommand(CommandManager.kt) 2020-12-21 14:05:12 W/stderr:   at net.mamoe.mirai.console.terminal.ConsoleThreadKt$startupConsoleThread$3.invokeSuspend(ConsoleThread.kt:183) 2020-12-21 14:05:12 W/stderr:   at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33) 2020-12-21 14:05:12 W/stderr:   at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106) 2020-12-21 14:05:12 W/stderr:   at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571) 2020-12-21 14:05:12 W/stderr:   at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:738) 2020-12-21 14:05:12 W/stderr:   at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678) 2020-12-21 14:05:12 W/stderr:   at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665) 2020-12-21 14:05:12 W/stderr: Caused by: java.util.jar.JarException: file:/C:/Users/Administrator/Desktop/mirai/libs/mirai-core-all-2.0-M1-1-all.jar has unsigned entries - net/mamoe/mirai/internal/AbstractBot$2.class 2020-12-21 14:05:12 W/stderr:   at java.base/javax.crypto.JarVerifier.verifySingleJar(JarVerifier.java:459) 2020-12-21 14:05:12 W/stderr:   at java.base/javax.crypto.JarVerifier.verifyJars(JarVerifier.java:314) 2020-12-21 14:05:12 W/stderr:   at java.base/javax.crypto.JarVerifier.verify(JarVerifier.java:257) 2020-12-21 14:05:12 W/stderr:   at java.base/javax.crypto.ProviderVerifier.verify(ProviderVerifier.java:129) 2020-12-21 14:05:12 W/stderr:   at java.base/javax.crypto.JceSecurity.verifyProvider(JceSecurity.java:191) 2020-12-21 14:05:12 W/stderr:   at java.base/javax.crypto.JceSecurity.getVerificationResult(JceSecurity.java:217) 2020-12-21 14:05:12 W/stderr:   at java.base/javax.crypto.JceSecurity.getInstance(JceSecurity.java:129) 2020-12-21 14:05:12 W/stderr:   ... 36 more 2020-12-21 14:05:12 I/Bot.xxxxxxxxxxxx: Logging in... 2020-12-21 14:05:12 I/Bot.xxxxxxxxxxxx: 需要滑动验证码 2020-12-21 14:05:12 I/Bot.xxxxxxxxxxxx: 请在任意浏览器中打开以下链接并完成验证码. 2020-12-21 14:05:12 I/Bot.xxxxxxxxxxxx: 完成后请输入任意字符 2020-12-21 14:05:12 I/Bot.xxxxxxxxxxxx: https://xxxxxxxxxxxxxxxxxxxxxxx............................. 14:05:12 LOGIN&gt; LOGIN&gt;  &gt; 1 2020-12-21 14:05:24 I/Bot.xxxxxxxxxxxx: 正在提交中... Login failed: Error(title=禁止登录, message=当前上网环境异常，请更换网络环境或在常用设备上登录或稍后再试。, errorInfo=) 2020-12-21 14:05:24 E/console: net.mamoe.mirai.network.WrongPasswordException: Error(title=禁止登录, message=当前上网环 境异常，请更换网络环境或在常用设备上登录或稍后再试。, errorInfo=) net.mamoe.mirai.network.WrongPasswordException: Error(title=禁止登录, message=当前上网环境异常，请更换网络环境或在常用设备上登录或稍后再试。, errorInfo=)         at net.mamoe.mirai.internal.network.QQAndroidBotNetworkHandler.closeEverythingAndRelogin(QQAndroidBotNetworkHandler.kt:185)         at net.mamoe.mirai.internal.network.QQAndroidBotNetworkHandler$closeEverythingAndRelogin$1.invokeSuspend(QQAndroidBotNetworkHandler.kt)         at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)         at kotlinx.coroutines.internal.ScopeCoroutine.afterResume(Scopes.kt:32)         at kotlinx.coroutines.AbstractCoroutine.resumeWith(AbstractCoroutine.kt:113)         at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)         at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)         at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)         at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:738)         at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)         at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665) &gt; login xxxxxxxxxxxx xxxxxxx 2020-12-21 14:11:42 I/Bot.xxxxxxxxxxxx: Logging in... 2020-12-21 14:11:42 I/Bot.xxxxxxxxxxxx: 需要滑动验证码 2020-12-21 14:11:42 I/Bot.xxxxxxxxxxxx: 请在任意浏览器中打开以下链接并完成验证码. 2020-12-21 14:11:42 I/Bot.xxxxxxxxxxxx: 完成后请输入任意字符 2020-12-21 14:11:42 I/Bot.xxxxxxxxxxxx: https://xxxxxxxxxxxxxxxxxxxxxxx............................. 14:11:42 LOGIN&gt; LOGIN&gt;  &gt; 2020-12-21 14:12:28 I/Bot.xxxxxxxxxxxx: 正在提交中... Login failed: Error(title=禁止登录, message=当前上网环境异常，请更换网络环境或在常用设备上登录或稍后再试。, errorInfo=) 2020-12-21 14:12:28 E/console: net.mamoe.mirai.network.WrongPasswordException: Error(title=禁止登录, message=当前上网环 境异常，请更换网络环境或在常用设备上登录或稍后再试。, errorInfo=) net.mamoe.mirai.network.WrongPasswordException: Error(title=禁止登录, message=当前上网环境异常，请更换网络环境或在常用设备上登录或稍后再试。, errorInfo=)         at net.mamoe.mirai.internal.network.QQAndroidBotNetworkHandler.closeEverythingAndRelogin(QQAndroidBotNetworkHandler.kt:185)         at net.mamoe.mirai.internal.network.QQAndroidBotNetworkHandler$closeEverythingAndRelogin$1.invokeSuspend(QQAndroidBotNetworkHandler.kt)         at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)         at kotlinx.coroutines.internal.ScopeCoroutine.afterResume(Scopes.kt:32)         at kotlinx.coroutines.AbstractCoroutine.resumeWith(AbstractCoroutine.kt:113)         at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)         at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)         at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)         at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:738)         at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)         at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)  
—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub, or unsubscribe.

--- a/mirai-core/src/commonMain/kotlin/network/protocol/packet/login/WtLogin.kt
+++ b/mirai-core/src/commonMain/kotlin/network/protocol/packet/login/WtLogin.kt
@@ -44,7 +44,7 @@ internal class WtLogin {
                         t193(ticket)
                         t8(2052)
                         t104(client.t104)
-                        t116(150470524, 66560)
+                        t116(client.miscBitMap, client.subSigMap)
                     }
                 }
             }
@@ -61,7 +61,7 @@ internal class WtLogin {
                         t2(captchaAnswer, captchaSign, 0)
                         t8(2052)
                         t104(client.t104)
-                        t116(150470524, 66560)
+                        t116(client.miscBitMap, client.subSigMap)
                     }
                 }
             }
@@ -79,7 +79,7 @@ internal class WtLogin {
                         writeShort(4) // count of TLVs, probably ignored by server?
                         t8(2052)
                         t104(client.t104)
-                        t116(150470524, 66560)
+                        t116(client.miscBitMap, client.subSigMap)
                         t401(MiraiPlatformUtils.md5(client.device.guid + "stMNokHgxZUGhsYp".toByteArray() + t402))
                     }
                 }
@@ -105,7 +105,7 @@ internal class WtLogin {
                         writeShort(6) // count of TLVs, probably ignored by server?TODO
                         t8(2052)
                         t104(client.t104)
-                        t116(150470524, 66560)
+                        t116(client.miscBitMap, client.subSigMap)
                         t174(EMPTY_BYTE_ARRAY)
                         t17a(9)
                         t197(byteArrayOf(0.toByte()))

